modules/mixcr/1.2/mixcr.smk

```diff
@@ -0,0 +1,320 @@
+    run_dict = {"IGH":"Ig", "IGK":"Ig", "IGL":"Ig", "TRA":"TCR","TRB":"TCR","TRD":"TCR","TRG":"TCR"}
+
+    rule _igblastn_run:
+        input:
+            fasta = rules._mixcr_to_fasta.output.fasta,
+            ig_db = reference_files("genomes/no_build/igblast/database/imgt_database.success")
+        output:
+            db = CFG["dirs"]["mixcr"] + "{seq_type}/{sample_id}/mixcr.{sample_id}.clonotypes.{chain}.igblastn.fmt7"
+        params:
+            receptor_type = lambda wildcards: run_dict[wildcards.chain],
+            aux = reference_files("downloads/igblast/optional_file/human_gl.aux"),
+            gdv = lambda wildcards: (reference_files("genomes/no_build/igblast/database/imgt_human_" + receptor_dict[wildcards.chain] + "_v.ndb")).replace(".ndb",""),
+            gdj = lambda wildcards: (reference_files("genomes/no_build/igblast/database/imgt_human_" + receptor_dict[wildcards.chain] + "_j.ndb")).replace(".ndb",""),
+            gdd = lambda wildcards: (reference_files("genomes/no_build/igblast/database/imgt_human_" + receptor_dict[wildcards.chain] + "_d.ndb")).replace(".ndb",""),
+            run_flags = CFG["options"]["igblast_run"]["run_flags"],
+            form = "7 std btop " + CFG["options"]["igblast_run"]["form"]
+        conda:
+            CFG["conda_envs"]["igblast"]
+        shell:
+            op.as_one_line("""
+            igblastn -query {input.fasta} -out {output.db} 
+            -ig_seqtype {params.receptor_type} -organism human 
+            -auxiliary_data {params.aux} 
+            -germline_db_V {params.gdv} -germline_db_J {params.gdj} -germline_db_D {params.gdd} 
+            {params.run_flags} -outfmt '{params.form}' -domain_system imgt
+            """)
+
+    rule _update_mixcr_results:
+        input:
+            db = rules._igblastn_run.output.db,
+            og_mixcr = CFG["dirs"]["mixcr"] + "{seq_type}/{sample_id}/mixcr.{sample_id}.clonotypes.{chain}.txt",
+            seq_info = rules._mixcr_to_fasta.output.seq_info,
+            script = CFG["igblast_scripts"]["igblastn2mixcr"]
+        output:
+            txt = CFG["dirs"]["mixcr"] + "{seq_type}/{sample_id}/mixcr.{sample_id}.clonotypes.{chain}.status.txt"
+        shell:
+            "{input.script} -d {input.db} -m {input.og_mixcr} -o {output.txt} -s {input.seq_info}"
+
+    rule _symlink_mixcr_update:
+        input:
+            txt = rules._update_mixcr_results.output.txt
+        output:
+            txt = CFG["dirs"]["outputs"] + "txt/{seq_type}/mixcr.{sample_id}.clonotypes.{chain}.status.txt"
+        run:
+            op.relative_symlink(input.txt, output.txt, in_module=True)
+
+# Symlinks the final output files into the module results directory (under '99-outputs/')
+rule _mixcr_output_txt:
+    input:
+        txt = rules._mixcr_run.output.txt,
+        report = rules._mixcr_run.output.report
+    output:
+        txt = CFG["dirs"]["outputs"] + "txt/{seq_type}/mixcr.{sample_id}.clonotypes.ALL.txt",
```

Comment:
IIRC MiXCR creates an empty `clonotypes.ALL.txt` file when you specify either `bcr` or `tcr` for `--receptor-type`, so I would say it makes sense to change this rule to symlink the individual chain files instead of `ALL.txt`. Or you can do both if you really want to.

Reply:
Changed `_mixcr_output_txt` input to a single `_mixcr_run` output file: `CFG["dirs"]["mixcr"] + "{seq_type}/{sample_id}/mixcr.{sample_id}.clonotypes.{chain}.txt"` and changed output to `CFG["dirs"]["outputs"] + txt/{seq_type}/mixcr.{sample_id}.clonotypes.{chain}.txt"`